sam.c

```diff
@@ -2399,6 +2546,10 @@ static int sam_parse_B_vals(char type, uint32_t n, char *in, char **end,
     }
 
     // Ensure space for type + values
+    if (!n) n=7;
+    uint32_t nused = 0;
+    int64_t nalloc = n;
```

Comment:
Instead of taking a copy of `n`, you could just make `nalloc` the input parameter.

Reply:
The sizes are different, but I assign from n to nalloc and back again so that wasn't valid anyway.  BAM can't cope with more than 32-bit, so I'll make nalloc uint32_t too.